openfeature/provider/in_memory_provider.py

```diff
@@ -117,3 +117,56 @@ def _resolve(
         if flag is None:
             raise FlagNotFoundError(f"Flag '{flag_key}' not found")
         return flag.resolve(evaluation_context)
+
+
+class AsyncInMemoryProvider(InMemoryProvider):
+    _flags: FlagStorage
+
+    def __init__(self, flags: FlagStorage) -> None:
+        self._flags = flags.copy()
+
+    def get_metadata(self) -> Metadata:
+        return InMemoryMetadata()
+
+    def get_provider_hooks(self) -> typing.List[Hook]:
+        return []
```

Comment:
maybe I'm wrong, but I think these two should also be `async` 🤔

Reply:
Do you mean more generally or specifically to the InMemoryProvider?

For this PR, I was limiting the scope of adding async functionality to the resolving calls. 
This ensures we have backwards compatibility with current hooks / metadata calls and providers would just need to implement the async resolve calls.

I do agree support async hooks and metadata retrieval should be looked at in the future though.

Reply:
Yeah, meant it more generally, because currently those would block the call, which is not really relevant for the in-memory variant, but would be for remote calls.

But totally fine to change this in the future.